src/movies.js

```diff
@@ -1,16 +1,140 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
-
+function getAllDirectors (arr) {
+    let output = [];
+    let allDirs = arr.forEach((movie)=>{
+      //console.log(obj.director)
+       output.push(movie.director);
+     })
+    return output.reduce((acc,currIndex) => {
+    if (acc.includes(currIndex)) {
+      return acc;
+    } else {
+      return [...acc, currIndex];
+    }
+  }, []);
+  };
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+function howManyMovies(movies) {
+    const dramaSpielbergMovies = movies.filter((movie) => {
+       return movie.genre.includes("Drama") && movie.director === "Steven Spielberg";
+    })
+    return dramaSpielbergMovies.length;
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
-
```

Comment:
Nice Elia!

Here's a different but similar solution, just to give you another perspective
`function ratesAverage(lotsOfMovies) {

if (!lotsOfMovies.length) {
return 0;
}

let total = lotsOfMovies.reduce((a, b) => {
if (b.rate) {
return a + b.rate;
} else {
return a;
}
}, 0);

// you can use Number(), parseInt() or simply plus +
return Number((total / lotsOfMovies.length).toFixed(2));
}`